modules/apps/data-engine/data-engine-taglib/test/js/data-layout-builder/components/rule-builder/editor/Editor.es.js

```diff
@@ -0,0 +1,833 @@
+						await waitForElement(() => {
+							return document.querySelector(
+								'.dropdown-menu.show'
+							);
+						});
+
+						await act(async () => {
+							fireEvent.click(getByText(firstArg));
+						});
+
+						expect(getByText('Is equal to')).toBeTruthy();
+						expect(getByText('Is not equal to')).toBeTruthy();
+						expect(getByText('Is greater than')).toBeTruthy();
+						expect(
+							getByText('Is greater than or equal to')
+						).toBeTruthy();
+						expect(getByText('Is less than')).toBeTruthy();
+						expect(getByText('Is not empty')).toBeTruthy();
+						expect(getByText('Is empty')).toBeTruthy();
+						expect(getByText('Is not empty')).toBeTruthy();
+					}
+				);
+			});
+
+			describe('Binary operations', () => {
+				it.each(
+					STRING_DATATYPE_FIELDS.concat(NUMBER_TYPE_FIELDS).concat(
+						UPLOAD_TYPE_FIELD
+					)
+				)(
+					'shows a related field input on the right operand when the left operand is a %p and action type is value',
+					async (firstArg) => {
+						const props = defaultProps();
+						const mockIsSignedIn = jest.fn();
+
+						Liferay.ThemeDisplay.isSignedIn = mockIsSignedIn;
+						const {getByText} = render(
+							<Editor
+								{...props}
+								onChange={() => {}}
+								onValidator={() => {}}
+							/>
+						);
+
+						await waitForElement(() => {
+							return document.querySelector('.option-selected');
+						});
+
+						const fieldLeft = document
+							.querySelectorAll('.timeline-item')[1]
+							.querySelectorAll('.ddm-field')[0]
+							.querySelector('.option-selected');
+
+						await act(async () => {
+							fireEvent.click(fieldLeft);
+						});
+
+						await waitForElement(() => {
+							return document.querySelector(
+								'.dropdown-menu.show'
+							);
+						});
+						await act(async () => {
+							fireEvent.click(getByText(firstArg));
+						});
+
+						const operator = document
+							.querySelectorAll('.timeline-item')[1]
+							.querySelectorAll('.ddm-field')[1]
+							.querySelector('.option-selected');
+
+						await act(async () => {
+							fireEvent.click(operator);
+						});
+
+						await act(async () => {
+							fireEvent.click(getByText('Is equal to'));
+						});
+
+						const actionsType = document
+							.querySelectorAll('.timeline-item')[1]
+							.querySelectorAll('.ddm-field')[2]
+							.querySelector('.option-selected');
+
+						fireEvent.click(actionsType);
+
+						fireEvent.click(getByText('value'));
+
+						await waitForElement(() => {
+							return document.querySelector('.ddm-field');
+						});
+
+						if (firstArg === 'text') {
+							expect(
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector('input.ddm-field-text')
+							).toBeTruthy();
+						}
+						else if (
+							firstArg === 'select' ||
+							firstArg == 'checkbox_multiple' ||
+							firstArg == 'radio'
+						) {
+							expect(
+								document
```

Comment:
Why not using [`(getBy*`](https://testing-library.com/docs/dom-testing-library/api-queries/)?